realm/realm-library/src/objectServer/java/io/realm/PermissionManager.java

```diff
@@ -0,0 +1,388 @@
+                        synchronized (errorLock) {
+                            managementRealmError = error;
+                        }
+                    }
+                })
+                .modules(new ManagementModule())
+                .build();
+
+        permissionRealmConfig = new SyncConfiguration.Builder(
+                user, getRealmUrl(RealmType.PERMISSION_REALM, user.getAuthenticationUrl()))
+                .errorHandler(new SyncSession.ErrorHandler() {
+                    @Override
+                    public void onError(SyncSession session, ObjectServerError error) {
+                        synchronized (errorLock) {
+                            permissionRealmError = error;
+                        }
+                    }
+                })
+                .modules(new PermissionModule())
+                .waitForInitialRemoteData()
+                .readOnly()
+                .build();
+    }
+
+    /**
+     * FIXME: Add Javadoc
+     *
+     * @param callback
+     * @return
+     */
+    public RealmAsyncTask getPermissionsAsync(final Callback<RealmResults<Permission>> callback) {
+        checkIfValidThread();
+        checkCallbackNotNull(callback);
+        return addTask(new PermissionsAsyncTask(callback));
+    }
+
+    // Queue the task if the underlying Realms are not ready yet, otherwise
+    // start the task by sending it to this thread handler. This is done
+    // in order to be able to provide the user with a RealmAsyncTask representation
+    // of the work being done.
+    private RealmAsyncTask addTask(final PermissionManagerAsyncTask task) {
+        if (isReady()) {
+            activateTask(task);
+        } else {
+            delayTask(task);
+            openRealms();
+        }
+
+        return task;
+    }
+
+    // Park the task until all underlying Realms are ready
+    private void delayTask(PermissionManagerAsyncTask task) {
+        delayedTasks.add(task);
+    }
+
+    // Run any tasks that where delayed while the underlying Realms where being opened.
+    // PRECONDITION: Underlying Realms are no longer in the process of being opened.
+    private void runDelayedTasks() {
+        for (PermissionManagerAsyncTask delayedTask : delayedTasks) {
+            activateTask(delayedTask);
+        }
+        delayedTasks.clear();
+    }
+
+    // Activate a task. All tasks are controlled by the Handler in order to make it asynchronous.
+    // PRECONDITION: Underlying Realms are no longer in the process of being opened.
+    private void activateTask(PermissionManagerAsyncTask task) {
+        activeTasks.add(task);
+        handler.post(task);
+    }
+
+    // Open both underlying Realms asynchronously. Once they are both ready, all tasks added in the meantime are
+    // started. If the Realms failed to open correctly any error will be reported through the `Callback.onError` callback.
+    private void openRealms() {
```

Comment:
Yes, this method is ever only be called on the thread that opened the PermissionManager